src/compiler/checker.ts

```diff
@@ -11749,23 +11749,23 @@ module ts {
             return !!resolveName(location, name, SymbolFlags.Value, /*nodeNotFoundMessage*/ undefined, /*nameArg*/ undefined);
         }
 
-        function getBlockScopedVariableId(n: Identifier): number {
+        function getNamedValueId(n: Identifier, meaning: SymbolFlags): number {

```

Comment:
Why is it necessary to take the specific meaning? Why can't this function simply always check for `SymbolFlags.BlockScopedVariable | SymbolFlags.Class`?